tests/unit/list/dlist.c

```diff
@@ -167,7 +169,185 @@ static inline bool verify_tail_head(sys_dlist_t *list,
  */
 
 /**
- * @brief Verify doubly linked list funtionalities
+ * @brief Test dlist get head and tail operation
+ *
+ * @details
+ * Initialize a dlist and append it to make it have different nodes
+ * Compare the time spent of get head operation
+ * Compare the time spent of get tail operation
+ *
+ * @ingroup lib_dlist_tests
+ *
+ * @see sys_dlist_peek_head(),sys_dlist_peek_tail()
+ */
+void test_dlist_peek_head_tail(void)
+{
+	int i;
+	/*dlist node is embedded in user struct*/
+	struct container_node *cnode;
+	const int node_size = 20;
+	u32_t start_time_1;
+	u32_t const_time[6];
+	struct container_node data_node[node_size];
+	sys_dlist_t test_list;
+
+	/* Initialize dlist*/
+	sys_dlist_init(&test_list);
+
+	for (i = 0; i < node_size; i++) {
+		data_node[i].unused = i + 1;
+	}
+
+	/*for-loop to make the dlist of different sizes*/
+	for (i = 0; i < 6; i++) {
+		sys_dlist_append(&test_list, &data_node[i].node);
+	}
```

Comment:
This seems clumsy, you're repeating effectively identical code again and again and again.  If the point is to check performance of lists of different sizes, please structure this as a separate function like `check_dlist_perf(size_t list_size, struct dlist_perf_stats *stats);`, with:

```c
struct dlist_perf_stats {
    u32_t peek_head_cyc;
    u32_t peek_tail_cyc;
    u32_t insert_cyc;
    u32_t remove_cyc;
};
```

Then just call it a bunch of times for lists of different sizes and do your statistics in one place.

Reply:
Hi @andyross ,thanks for your comments, I'm sorry to repeat effectively the same codes for more times, I totally approve that it's a good way to structure this as a separate function. I follow your methods above to modify my code style.

Unfortunately, I test this point basing on the zephyr platform. But the double list is a type of data structure, so it's necessary to separate from all platforms, as you see, a lot of changes have been made about my new testcase than before. And adding some code comments to help others to understand.